pandas/_libs/lib.pyx

```diff
@@ -1484,23 +1484,17 @@ def infer_dtype(value: object, skipna: bool = True) -> str:
 
     if util.is_array(value):
         values = value
-    elif hasattr(value, "inferred_type") and skipna is False:
+    elif hasattr(type(value), "inferred_type") and skipna is False:
```

Comment:
The issue  in the old version was that using `hasattr` on a `cache_readonly` is slow. By doing this we get a good speed-up.

Reply:
nice!